internal/common/disasters/commonpooldefinition.go

```diff
@@ -0,0 +1,56 @@
+package disasters
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+)
+
+type Commonpool struct {															
+	Resource, Threshold  uint
+}
+
+// DisasterMitigate mitigates the disaster's damage using CP before it hits the island
+// 	+ If cp have enough resource to fully mitigate the disaster, island's personal resources won't be affected 
+//	+ Else, island will be affected with leftover damaged. Each island receives proportional helps from cp with respect to the total damage on 6 islands 
+//	+ EX: if island 1 gets hit the most then cp will help island 1 more than the rest 
+
+//	+ There is a threshold T that incentivizes contributing to the cp from islands
+//	+ If threshold T is met, which mean islands have prepared well before the disaster hit, the damage that disaster have on cp and islands will be half 
+//	+ If not, cp and islands will feel the full effect of disaster
+//	+ For now, each 1 effect of disaster is equivalent to 1000 resource 
+func (e *Environment) DisasterMitigate(individualEffect map[shared.ClientID]float64, proportionalEffect map[shared.ClientID]float64) map[shared.ClientID]float64 {				//input is the porportional effect of each island
+	// compute total effect of disaster on 6 islands 
```

Comment:
As per previous requests, the server no longer stores an Environment pointed but rather just an instance of the struct itself.

If this is indeed the desired behaviour, we will need to modify the env copy in this function and return the updated env to be reassigned to the game state after I suppose?

Reply:
Hold up; So is this supposed to actually mutate the environment?

From what I can see, this function is a pure function and computes an update, so there's no reason for it to have a pointer receiver in this case.

Reply:
L52 mutates it. 

The server's containment of Environment does not matter--the same principle still applies: pointer receiver methods modify the caller, value receiver methods don't.

Reply:
I generally _always_ prefer reassignment structs after modifying them (no matter using pointer receiver methods or by getting a new value out of a value receiver), but this method works as well. Up to you.

Note that in Python, it's always implicitly pointer receiver for all objects.

Reply:
@Darrekt yes - that was what I was saying. That it shouldn't be a pointer receiver and we should use reassignment rather. I was just referring to server's containment as an example of previous time this came up.

Reply:
_immutable state_